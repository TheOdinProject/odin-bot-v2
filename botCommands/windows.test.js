const command = require('./windows');

describe('?windows', () => {
  describe('regex', () => {
    it.each([
      ['?windows'],
      [' ?windows'],
      ['?windows @odin-bot'],
      ['@odin-bot ?windows'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['wndows'],
      ['windows'],
      ['window'],
      ['?wndows'],
      ['?window'],
      ['? windows'],
      ['?awindows'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! ?windows'],
      ['Don\'t worry about ?windows'],
      ['Hey @odin-bot, ?windows'],
      ['?@odin-bot ^ ?me ?windows ?tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user?windows'],
      ['it\'s about?windows'],
      ['?windowsanillusion'],
      ['?windows?'],
      ['?windows*'],
      ['?windows...'],
    ])("'%s' - command should be its own word/group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(command.cb()).toMatchSnapshot();
    });
  });
});
