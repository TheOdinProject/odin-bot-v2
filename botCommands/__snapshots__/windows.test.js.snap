// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!windows callback returns correct output 1`] = `
Object {
  "embeds": Array [
    Object {
      "color": 13407555,
      "description": "**The Odin Project does not support Windows, WSL, or any OS outside of our recommendations**. We are happy to assist with any questions about installing a VM or dual booting Linux. <https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/installation-overview#os-options>",
      "title": "Windows",
      "url": "https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/installation-overview#os-options",
    },
  ],
}
`;
